File: ballast-client/npm/src/signalr/signalr-client-game-service.ts
import { IAddPlayerOptions, ICreateGameOptions, IGameDto, IGameService, IRemovePlayerOptions, IVesselDto, IVesselMoveRequest } from "ballast-core";

export class SignalRClientGameService implements IGameService {
    dispose(): void {
        throw new Error("Method not implemented.");
    }
    getAllGamesAsync(): Promise<IGameDto[]> {
        throw new Error("Method not implemented.");
    }    
    getGameAsync(gameId: string): Promise<IGameDto> {
        throw new Error("Method not implemented.");
    }
    getTestGameIdAsync(): Promise<string> {
        throw new Error("Method not implemented.");
    }
    createGameAsync(options: ICreateGameOptions): Promise<IGameDto> {
        throw new Error("Method not implemented.");
    }
    startGameAsync(gameId: string): Promise<IGameDto> {
        throw new Error("Method not implemented.");
    }
    endGameAsync(gameId: string): Promise<IGameDto> {
        throw new Error("Method not implemented.");
    }
    deleteGameAsync(gameId: string): Promise<void> {
        throw new Error("Method not implemented.");
    }
    addPlayerToGameAsync(options: IAddPlayerOptions): Promise<IGameDto> {
        throw new Error("Method not implemented.");
    }
    removePlayerFromGameAsync(options: IRemovePlayerOptions): Promise<IGameDto> {
        throw new Error("Method not implemented.");
    }
    addPlayerToVesselAsync(options: IAddPlayerOptions): Promise<IVesselDto> {
        throw new Error("Method not implemented.");
    }
    removePlayerFromVesselAsync(options: IRemovePlayerOptions): Promise<IVesselDto> {
        throw new Error("Method not implemented.");
    }
    addPlayerToVesselRoleAsync(options: IAddPlayerOptions): Promise<IVesselDto> {
        throw new Error("Method not implemented.");
    }
    removePlayerFromVesselRoleAsync(options: IRemovePlayerOptions): Promise<IVesselDto> {
        throw new Error("Method not implemented.");
    }
    moveVesselAsync(request: IVesselMoveRequest): Promise<IVesselDto> {
        throw new Error("Method not implemented.");
    }
}

File: ballast-client/dotnet/src/signalr/SignalRClientGameService.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Ballast.Core.Application.Models;
using Ballast.Core.Application.Services;

namespace Ballast.Client.SignalR
{
    public class SignalRClientGameService : IGameService
    {
        
        public void Dispose()
        {
            throw new NotImplementedException();
        }

        public Task<GameDto> AddPlayerToGameAsync(AddPlayerOptions options)
        {
            throw new NotImplementedException();
        }

        public Task<VesselDto> AddPlayerToVesselAsync(AddPlayerOptions options)
        {
            throw new NotImplementedException();
        }

        public Task<VesselDto> AddPlayerToVesselRoleAsync(AddPlayerOptions options)
        {
            throw new NotImplementedException();
        }

        public Task<GameDto> CreateGameAsync(CreateGameOptions options)
        {
            throw new NotImplementedException();
        }

        public Task DeleteGameAsync(Guid gameId)
        {
            throw new NotImplementedException();
        }

        public Task<GameDto> EndGameAsync(Guid gameId)
        {
            throw new NotImplementedException();
        }

        public Task<IEnumerable<GameDto>> GetAllGamesAsync()
        {
            throw new NotImplementedException();
        }

        public Task<GameDto> GetGameAsync(Guid gameId)
        {
            throw new NotImplementedException();
        }

        public Task<Guid> GetTestGameIdAsync()
        {
            throw new NotImplementedException();
        }

        public Task<VesselDto> MoveVesselAsync(VesselMoveRequest request)
        {
            throw new NotImplementedException();
        }

        public Task<GameDto> RemovePlayerFromGameAsync(RemovePlayerOptions options)
        {
            throw new NotImplementedException();
        }

        public Task<VesselDto> RemovePlayerFromVesselAsync(RemovePlayerOptions options)
        {
            throw new NotImplementedException();
        }

        public Task<VesselDto> RemovePlayerFromVesselRoleAsync(RemovePlayerOptions options)
        {
            throw new NotImplementedException();
        }

        public Task<GameDto> StartGameAsync(Guid gameId)
        {
            throw new NotImplementedException();
        }
    }
}